integration_tests/benchmarks/object-detection/benchmark_script.py

```diff
@@ -232,61 +173,226 @@ def run_detailed_pr_curve_evaluation(dset: Dataset, model: Model):
-
-        try:
-            eval_ = run_base_evaluation(dset=dset, model=model)
-        except TimeoutError:
-            raise TimeoutError(
-                f"Evaluation timed out when processing {limit} datums."
-            )
-
-        start = time()
-        client.delete_dataset(dset.name, timeout=30)
-        client.delete_model(model.name, timeout=30)
-        deletion_time = time() - start
-
-        results = {
-            "number_of_datums": limit,
-            "number_of_unique_labels": eval_.meta["labels"],
-            "number_of_annotations": eval_.meta["annotations"],
-            "ingest_runtime": f"{(ingest_time):.1f} seconds",
-            "eval_runtime": f"{(eval_.meta['duration']):.1f} seconds",
-            "del_runtime": f"{(deletion_time):.1f} seconds",
-        }
-        write_results_to_file(write_path=write_path, result_dict=results)
+        for gt_type, gt_filename in groundtruths.items():
+            for pd_type, pd_filename in predictions.items():
+
+                print(gt_type, pd_type)
+
+                try:
+                    dataset = Dataset.create(name="coco")
+                    model = Model.create(name="yolo")
+                except (
+                    DatasetAlreadyExistsError,
+                    ModelAlreadyExistsError,
+                ) as e:
+                    client.delete_dataset("coco")
+                    client.delete_model("yolo")
+                    raise e
+
+                # gt bbox ingestion
+                gt_ingest_time = time_it(
+                    ingest_groundtruths,
+                    dataset=dataset,
+                    path=current_directory / Path(gt_filename),
+                    limit=limit,
+                    chunk_size=1000,
+                )
+
+                # gt bbox finalization
+                gt_finalization_time = time_it(dataset.finalize)
+
+                # pd bbox ingestion
+                box_datum_uids = [datum.uid for datum in dataset.get_datums()]
+                pd_ingest_time = time_it(
+                    ingest_predictions,
+                    dataset=dataset,
+                    model=model,
+                    datum_uids=box_datum_uids,
+                    path=current_directory / Path(pd_filename),
+                    limit=limit,
+                    chunk_size=1000,
+                )
+
+                # model finalization
+                pd_finalization_time = time_it(
+                    model.finalize_inferences, dataset
+                )
+
+                try:
+                    eval_base = run_base_evaluation(dset=dataset, model=model)
+                    eval_pr = run_pr_curve_evaluation(
+                        dset=dataset, model=model
+                    )
+                    eval_detail = run_detailed_pr_curve_evaluation(
+                        dset=dataset, model=model
+                    )
+                except TimeoutError:
+                    raise TimeoutError(
```

Comment:
will this catch work since you're removing timeout=60 from wait_for_completion? back in the day, we'd said that we wanted to have this benchmark script fail if an individual evaluation took more than 30 seconds. it'd be ideal to return to that standard if possible

Reply:
I just reworked how timeouts work. By default it should fail at 30s now. 

The timeouts for both ingestion and evaluation can be adjusted in the args of `run_benchmarking_analysis`